addons/mail/models/res_users.py

```diff
@@ -64,10 +69,13 @@ def _compute_notification_type(self):
         new_portal_users.notification_type = 'email'
         new_portal_users.write({"group_ids": [Command.unlink(inbox_group_id)]})
 
-    @api.depends("presence_ids.status")
+    @api.depends("manual_im_status", "presence_ids.status")
     def _compute_im_status(self):
         for user in self:
-            user.im_status = user.presence_ids.status or "offline"
+            user.im_status = (
+                "offline" if user.presence_ids.status == "offline" else
+                user.manual_im_status or user.presence_ids.status or "offline"
```

Comment:
the final `offline` which it meant for "no presence" should take priority over the manual status I think.